zanata-war/src/main/java/org/zanata/service/impl/TranslationStateCacheImpl.java

```diff
@@ -191,18 +191,22 @@ public Boolean textFlowTargetHasWarningOrError(Long targetId,
     // TODO why not @Observe the event directly?
     @Override
     public void textFlowStateUpdated(TextFlowTargetStateEvent event) {
-        DocumentLocaleKey key =
-                new DocumentLocaleKey(event.getDocumentId(),
-                        event.getLocaleId());
+        Long documentId = event.getKey().getDocumentId();
+        LocaleId localeId = event.getKey().getLocaleId();
 
         //invalidate document statistic cache
-        clearDocumentStatistics(event.getDocumentId(), event.getLocaleId());
+        clearDocumentStatistics(documentId, localeId);
 
-        // update document status information
-        updateDocStatusCache(key, event.getTextFlowTargetId());
+        for(TextFlowTargetStateEvent.TextFlowTargetState state: event.getStates()) {
+            Long tftId = state.getTextFlowTargetId();
 
-        // invalidate target validation
-        targetValidationCache.remove(event.getTextFlowTargetId());
+            // update document status information
+            updateDocStatusCache(new DocumentLocaleKey(documentId,
+                localeId), tftId);
```

Comment:
We need to pass in tftId into updateDocStatusCache. The cache needs to update `lastModifiedBy` information for that document by using target.getLastmodified.